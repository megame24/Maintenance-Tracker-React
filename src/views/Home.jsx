import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Banner, HomeBody, HomeFooter } from '../components/helpers/HomeHelper';

export const Home = ({ token }) => (
  <section id="landing">
    <Banner token={token} />
    <div className="container">
      <HomeBody />
      <hr />
      <HomeFooter token={token} />
    </div>
  </section>
);

Home.propTypes = {
  token: PropTypes.string,
};

Home.defaultProps = {
  token: ''
};

export const mapStateToProps = ({
  auth: { token }
}) => ({
  token,
});

export default connect(mapStateToProps)(Home);
