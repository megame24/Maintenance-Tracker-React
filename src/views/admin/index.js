import Dashboard from './Dashboard';
import AdminRequestDetails from './AdminRequestDetails';

export default {
  Dashboard,
  AdminRequestDetails,
};
