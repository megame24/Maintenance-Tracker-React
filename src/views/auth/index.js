import Signup from './Signup';
import Login from './Login';

export default {
  Signup,
  Login,
};
