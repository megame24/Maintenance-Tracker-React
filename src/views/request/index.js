import CreateRequest from './CreateRequest';
import ViewRequests from './ViewRequests';
import UserRequestDetails from './UserRequestDetails';
import EditRequest from './EditRequest';

export default {
  CreateRequest,
  ViewRequests,
  UserRequestDetails,
  EditRequest,
};
