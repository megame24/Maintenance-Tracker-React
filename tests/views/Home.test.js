import React from 'react';
import { shallow } from "enzyme";
import { Home, mapStateToProps } from '../../src/views/Home';


describe("Testing mapStateToProps", () => {
  it("should map the state to the props correctly", () => {
    const auth = {
      token: ''
    };
    const state = { auth };
    const componentState = mapStateToProps(state);
    expect(componentState).toEqual(auth);
  });
});

describe('Testing Home component', () => {
  it('should render as expected', () => {
    const tree = shallow(<Home />);
    expect(tree).toMatchSnapshot();
  });
});
