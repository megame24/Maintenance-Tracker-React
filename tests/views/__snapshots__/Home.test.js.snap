// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home component should render as expected 1`] = `
<section
  id="landing"
>
  <Banner
    token=""
  />
  <div
    className="container"
  >
    <HomeBody />
    <hr />
    <HomeFooter
      token=""
    />
  </div>
</section>
`;
